addons/stock_picking_batch/tests/test_wave_picking.py

```diff
@@ -553,3 +553,52 @@ def test_validatation_of_partially_empty_picking(self):
         self.assertEqual(wave.picking_ids, picking_1)
         self.assertEqual([picking_1.state, picking_1.move_ids.quantity, picking_1.move_ids.picked], ['done', 2.0, True])
         self.assertEqual([picking_2.state, picking_2.move_ids.quantity, picking_2.move_ids.picked], ['assigned', 0.0, True])
+
+    def test_add_partially_assigned_move_to_batch(self):
+        """
+            Checks that a picking is linked to the wave transfer in case all of its
+            moves are to be linked with the wave transfer.
+        """
+        picking = self.picking_internal
+        # update a move for the moved qty to be less than the initial demand
+        picking.move_ids[0].quantity = 10.0
+        self.assertRecordValues(picking.move_ids, [{'product_uom_qty': 15.0, 'quantity': 10.0}, {'product_uom_qty': 5.0, 'quantity': 5.0}])
+        lines = picking.move_ids.move_line_ids
+        res_dict = lines.action_open_add_to_wave()
+        res_dict['context'] = {'active_model': 'stock.move.line', 'active_ids': lines.ids}
+        wizard_form = Form(self.env[res_dict['res_model']].with_context(res_dict['context']))
+        wizard_form.mode = 'new'
+        wizard_form.save().attach_pickings()
+        # check that the picking was added to the wave transfer
+        wave = picking.batch_id.filtered(lambda b: b.is_wave)
+        self.assertTrue(wave)
+        # check that the lines are still linked to the original picking
+        self.assertEqual(lines.move_id.picking_id, picking)
+        # check that no other picking was added to the wave transfer
+        self.assertEqual(wave.move_ids, picking.move_ids)
+
+    def test_dont_add_empty_move_to_batch(self):
```

Comment:
This test does not crash without your fix. Could you update it please ?

Reply:
Hello @DjamelTouati indeed it does not crashes but I added this second test to be sure that my fix did not break the fix made by commit 8d0e49c (which didn't add a test).